### PR TITLE
Upgrade Ruby to 2.7.5

### PR DESCRIPTION
**Pull request by em-cd**
Upgrades Ruby to 2.7.5 and fixes some deprecation warnings in prep for Ruby 3.

This also includes:
* Changes to our Github workflow to use [ruby/setup-ruby](https://github.com/ruby/setup-ruby) and use its bundler cache (this sped up the Ruby setup part of our workflow by about 95% 🤯 )
* A few new tests to keep Codecov happy 🎁 🎄

### CODE REVIEW

File: app/views/admin/ofsted/_shortcut-nav.html.erb [hunk 1]
```diff
@@ -1,31 +1,34 @@
...
+
     <%= render partial: "shared/shortcuts_item", 
-    locals: {
-        state: current_page?(controller: 'admin/ofsted', action: 'pending') , 
-        link: pending_admin_ofsted_index_path,
-        count: @ofsted_counts.dig(:all, :pending),
-        label: "Pending"
-    } %>
+        locals: {
+            state: current_page?({controller: 'admin/ofsted', action: 'index'}) && !params[:archived].present?,
```
Comment (em-cd) [inline app/views/admin/ofsted/_shortcut-nav.html.erb:5]: This diff is really ugly because I fixed some indentation while I was here, but the main change is the added curly braces in the `current_page?` method calls in prep for [Ruby 3's positional and keyword arg changes](https://www.ruby-lang.org/en/news/2019/12/12/separation-of-positional-and-keyword-arguments-in-ruby-3-0/)

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -35,24 +35,10 @@ jobs:
         run: |
           sudo apt-get -y install libpq-dev
 
-      - name: Set up Ruby 2.6.0
-        uses: actions/setup-ruby@v1
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@v1
         with:
-          ruby-version: 2.6
```
Comment (em-cd) [inline .github/workflows/test.yml:41]: We can get rid of this because [ruby/setup-ruby](https://github.com/ruby/setup-ruby) uses our `.ruby-version` instead 😄

File: db/schema.rb [hunk 3]
```diff
@@ -13,7 +13,6 @@
 ActiveRecord::Schema.define(version: 2021_09_14_144638) do
 
   # These are extensions that must be enabled in order to support this database
-  enable_extension "pg_stat_statements"
```
Comment (em-cd) [inline db/schema.rb:16]: This line seemed to have been introduced in a commit with unrelated changes. We don't seem to be using it and it was causing problems, so I've removed it.

File: .github/workflows/test.yml (hunk 2 above)
Comment (apricot13) [reply .github/workflows/test.yml:41]: much easier!

**Review by apricot13 — APPROVED** [review]
The demo branch (also on this new ruby version) is running on Heroku like a charm as well!
